File: 栈/1910. Remove All Occurrences of a Substring.cpp
class Solution {
public:
    string removeOccurrences(string s, string part) {
        string ans;
        const int n = s.size(), m = part.size();
        for (int i = 0; i < n; i++) {
            ans += s[i];
            while (ans.size() >= m && ans.substr(ans.size() - m, m) == part) {
                for (int j = 0; j < m; j++) ans.pop_back();
            }
        }
        return ans;
    }
};

File: 字符串/1910. Remove All Occurrences of a Substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
 var removeOccurrences = function(s, part) {
    while (s.includes(part)) {
        s = s.replace(part, "");
    }
    return s;
};